### PR TITLE
Add the "writing a good package page" page

### PR DESCRIPTION
**Pull request by jayoung-lee**
Staged: https://dartlang-org-staging-0.firebaseapp.com/guides/libraries/writing-package-pages

### CODE REVIEW

File: src/_guides/libraries/writing-package-pages.md
Comment (parlough) [inline src/_guides/libraries/writing-package-pages.md:21]: I'm curious if it is worth mentioning that it may be better to not include a Markdown title in the README as otherwise it will appear multiple times on pub.dev. Many of the Dart team's packages previously removed theirs as a result of this. An example of a package that still does include a Markdown title though is [dartdoc](https://pub.dev/packages/dartdoc). There is an [pub.dev issue](https://github.com/dart-lang/pub-dev/issues/3238) tracking automatic removal of these, but it seems to be on hold.

image

File: src/_guides/libraries/writing-package-pages.md
Comment (parlough) [inline src/_guides/libraries/writing-package-pages.md:61]: Perhaps a generic term like "video" rather than "mov" would be better, since more people will likely understand and I believe at least GitHub supports other container formats besides mov. Both are fine though :)

**Comment by kwalrath** [comment]
I've just staged this: https://dartlang-org-staging-0.firebaseapp.com/guides/libraries/writing-package-pages

**Comment by kwalrath** [comment]
@jayoung-lee could you please produce a side-by-side figure for https://dartlang-org-staging-0.firebaseapp.com/guides/libraries/writing-package-pages#2-include-visual-content-to-explain-things? Currently the two screenshots show up one after the other (and huge!), but the text talks about the left and right figures (which would be better).

**Comment by jayoung-lee** [comment]
I added a side-by-side figure and deleted the old ones. Hope this works!

Is there a way to make all the images to be like 70-80% of the page width? Other images look very big too.

**Comment by kwalrath** [comment]
> I added a side-by-side figure and deleted the old ones. Hope this works!

Yes, much better!
 
> Is there a way to make all the images to be like 70-80% of the page width? Other images look very big too.

I'll try a few things and let you know how it goes.

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [reply src/_guides/libraries/writing-package-pages.md:36]: This is mentioned at the end of the first paragraph of tip 1, but it's easy to miss. I'll think about what to do...

**Comment by kwalrath** [comment]
I've staged the new image styling: https://dartlang-org-staging-0.firebaseapp.com/guides/libraries/writing-package-pages

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [inline src/_guides/libraries/writing-package-pages.md:25]: Can you redo this figure? Two reasons:

* It doesn't include the title **url_launcher** in the area of **This document**, but that text _is_ provided by the README.
* It shows the title **url_launcher**. 😆  We probably want to show a better example, no?

Could we use in_app_purchase, yaml, or another nicely done package page?

Comment (kwalrath) [reply src/_guides/libraries/writing-package-pages.md:25]: I wonder if we should change "This document" to "README file (this document)"

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [inline src/_guides/libraries/writing-package-pages.md:35]: Let's put the description down here, closer to the links, and make the link text match the figure labels exactly:

```suggestion
For details about other parts of the package page,
follow these links:

1. [Package layout ](https://dart.dev/tools/pub/package-layout)
2. [Flutter Favorite](https://flutter.dev/docs/development/packages-and-plugins/favorites)
3. [Package scoring](https://pub.dev/help/scoring)
4. [Verified publishers](https://dart.dev/tools/pub/verified-publishers)
5. [Pubspec file](https://dart.dev/tools/pub/pubspec)
```

**Review by kwalrath — COMMENTED** [review]
I'm going to have a bunch of small copy edits. @jayoung-lee are you OK with me just making commits, or would you like to review the changes?

File: src/_guides/libraries/writing-package-pages.md
Comment (jayoung-lee) [reply src/_guides/libraries/writing-package-pages.md:35]: Yes! Making commits sounds good to me. Thank you :D

File: src/_guides/libraries/writing-package-pages.md
Comment (jayoung-lee) [reply src/_guides/libraries/writing-package-pages.md:25]: The label "this document" is updated. And I replaced the image with `shared_preferences` page, which has a title `Shared preferences plugin`. Would this work?

Otherwise, we'll have to update one of the package's readme and retry. All the 1P Flutter Favorite packages have some sort of titles in the body, ironically :P

Comment (kwalrath) [reply src/_guides/libraries/writing-package-pages.md:25]: I'll take another look Monday

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [inline src/_guides/libraries/writing-package-pages.md:109]: Why are they better? Because they grab users' attention? Because users prefer them? Because they can convey more? (I'd like to be more specific here.)

Also, might you use an MP4 instead of a MOV?

**Comment by kwalrath** [comment]
@jayoung-lee I'm working on this today... and will have questions. :)

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [reply src/_guides/libraries/writing-package-pages.md:109]: I'm thinking of saying something like this:

>  Tip: Prefer animated GIFs and videos, when you have a choice. They can be more effective than static images for attracting readers and conveying information.

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [inline src/_guides/libraries/writing-package-pages.md:101]: This makes it seem like visual content is only useful for UI-related plugins. I'll think about how to fix it...

File: src/_guides/libraries/writing-package-pages.md
Comment (jayoung-lee) [reply src/_guides/libraries/writing-package-pages.md:109]: I like the new phrasing! 

Yeah, I'd say that UIs are not static in most cases (they're animated or they respond to user input), so videos or gifs are better in those cases, in terms of conveying information about the package. MP4 works too. :)

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [inline src/_guides/libraries/writing-package-pages.md:164]: Yikes, these links aren't working!

Comment (jayoung-lee) [reply src/_guides/libraries/writing-package-pages.md:164]: @jonasfj Do you know when the new version is going to be published?

Comment (kwalrath) [reply src/_guides/libraries/writing-package-pages.md:164]: Filed https://github.com/dart-lang/async/issues/168

Comment (jonasfj) [reply src/_guides/libraries/writing-package-pages.md:164]: I think we just need to publish it...

File: src/_guides/libraries/writing-package-pages.md
Comment (parlough) [inline src/_guides/libraries/writing-package-pages.md:91]: It's a plane...it's super man...it's a round hummingbird...it's dash!

File: src/_guides/libraries/writing-package-pages.md
Comment (parlough) [inline src/_guides/libraries/writing-package-pages.md:129]: Maybe instead of saying it looks `better`, we could be a bit more specific. Something more targetted such as `more informative`, `more fleshed out`, `more engaging`, or similar (I'm not really sure) to better indicate how it is "better".

File: src/_guides/libraries/writing-package-pages.md
Comment (kwalrath) [reply src/_guides/libraries/writing-package-pages.md:91]: It's Mega-Dash! 😆

**Comment by kwalrath** [comment]
I'm done with my copy edit, although I haven't yet addressed all of @parlough feedback. @jayoung-lee could you please take a look at my changes and at Parker's remaining comments? I've updated the staging server:

https://dartlang-org-staging-0.firebaseapp.com/guides/libraries/writing-package-pages

File: src/_guides/libraries/writing-package-pages.md
Comment (jayoung-lee) [reply src/_guides/libraries/writing-package-pages.md:129]: What about "look informative at first glance"? :)

**Comment by jayoung-lee** [comment]
LGTM! 

I made just one commit https://github.com/dart-lang/site-www/pull/3147/commits/d45ed4af9344e4714f3fdfaee73c9d485491b54d to address the comment from @parlough. I'll remove the `[PENDING]...` line, once https://github.com/dart-lang/async/issues/168 is closed.

Let me know if I missed anything.

**Comment by kwalrath** [comment]
LGTM once we can remove that PENDING comment!

**Comment by parlough** [comment]
@kwalrath @jayoung-lee Looks like the links in the async README are working now with the latest publish: https://pub.dev/documentation/async/latest/

**Review by parlough — APPROVED** [review]

**Review by kwalrath — APPROVED** [review]

**Comment by kwalrath** [comment]
LGTM. We'll want to add links to this from other docs and from the sidenav, but that could be in a follow-up PR. I'm thinking that https://dart.dev/tools/pub/publishing and https://dart.dev/tools/pub/package-layout (at least) should link to this page.

As for the sidenav, maybe **Writing package pages** could go just above **Package reference**, under **Publishing packages**?

image

**Comment by jayoung-lee** [comment]
> We'll want to add links to this from other docs and from the sidenav, but that could be in a follow-up PR. I'm thinking that https://dart.dev/tools/pub/publishing and https://dart.dev/tools/pub/package-layout (at least) should link to this page.
> 
> As for the sidenav, maybe **Writing package pages** could go just above **Package reference**, under **Publishing packages**?

Both sound good to me :)

**Pull request merged**
